src/core/meshcop/meshcop_tlvs.hpp

```diff
@@ -2248,13 +2248,42 @@ class DiscoveryResponseTlv : public Tlv, public TlvInfo<Tlv::kDiscoveryResponse>
         }
     }
 
+    /**
+     * This method indicates whether or not the Commercial Commissioning Mode flag is set.
+     *
+     * @retval TRUE   If the Commercial Commissioning Mode flag is set.
+     * @retval FALSE  If the Commercial Commissioning Mode flag is not set.
+     *
+     */
+    bool IsCommercialCommissioningMode(void) const { return (mFlags & kCCMMask) != 0; }
```

Comment:
```suggestion
    bool IsCommercialCommissioningMode(void) const { return !(mFlags & kCCMMask); }
```
The CCM flag has reverse meaning - `1` indicates that CCM mode is disabled.

Reply:
That's the case for Security Policy TLV only. For Discovery Request TLV:

> Its value is taken as the negated CCM bit value from the Security Policy TLV.

Reply:
The three negated bits are so confusing:
- kCommercialCommissioningMask
- kAutonomousEnrollmentMask
- kNetworkKeyProvisioningMask

I think we should rename them to be like `kNoCommercialCommissioningMask` or `kNegatedCommercialCommissioningMask`

Thoughts?

Reply:
@simonlingoogle fyi, kangping should have hidden the negated bits for security policy flags already in #6320, like
https://github.com/openthread/openthread/blob/4484042ced953b9d24a47cc8f3a04ea0e531e18c/src/core/thread/key_manager.cpp#L91

Reply:
@edmont Thanks for clarification! I missed the fact that the code is for the `Discovery Response TLV`.